include/rocksdb/merge_operator.h

```diff
@@ -222,26 +222,38 @@ class MergeOperator {
   }
 };
 
-// The simpler, associative merge operator.
+// A merge operator that can be implemented as an associative binary
+// operator on values/operands. Implementations must satisfy
+//      merge_two_values(merge_two_values(a, b), c)
+//   == merge_two_values(a, merge_two_values(b, c))
+// as partial merges might substitute the latter for the former.
+//
+// A consequence of this assumption is that values must be compatible as
+// operands in associative merge operators. For example, a silly general merge
```

Comment:
```
// A consequence of this assumption is that operands to an associative merge operator
// must also be valid values for storing into the database.
```

Reply:
That is not strictly true. It is the reverse that is strictly true (values must be valid as operands). My associative merge operation could be trim-whitespace-and-concatenate. I can assume values have trimmed whitespace, but operands might not yet be trimmed.

Reply:
That is true. I felt that it was easier to explain and understand the other way around... although I now see that is less accurate... we need some middle route ;-)

Reply:
And I think emphasizing the simpler IdentityAssocMergeOperator means we can brush all the weird nuance of AssociativeMergeOperator under the rug.